### PR TITLE
Set request Content-Type to application/json

### PR DESCRIPTION
**Pull request by richie3366**
In order to be compliant with the version 2 of the API (see https://api.random.org/json-rpc/2/fundamentals).
It is still compatible with the version 1 (see https://api.random.org/json-rpc/1/introduction).

### CODE REVIEW

**Comment by willfrew** [comment]
Hey @richie3366 thanks for the PR! The change itself LGTM but we'll need to fix up the [rpc tests](https://github.com/willfrew/node-random-org/blob/master/test/rpc.test.js).

Having had a quick look at the [output](https://travis-ci.org/willfrew/node-random-org/jobs/496682323), I think nock is probably pre-parsing the body as json given the changed content-type, resulting in [`requestBody`](https://github.com/willfrew/node-random-org/blob/master/test/rpc.test.js#L400) being an object rather than the plain string. Could you try telling nock not to parse the body, or getting the raw request body another way?

**Comment by richie3366** [comment]
The only workaround I found (even after searching for a way to get the raw body with nock) was to use `JSON.stringify` in the *Content-Length* test, and to remove the JSON.parse calls in the next failing tests.

```javascript
      var contentLength = Buffer.byteLength(JSON.stringify(requestBody), 'utf8');
      expect(request.headers['content-length']).to.equal(contentLength);
```

It works well, but I don't know if it is a legit & safe way to test it.
Please tell me what you think.

EDIT
Another way would be to only `JSON.stringify` here ; less changes to make.
```javascript
    .reply(200, function(uri, reqBody) {
      // Store the request object & body to assert about later.
      request = this.req;
      requestBody = JSON.stringify(reqBody)
      return jsonRpcResponse;
    });
```

**Comment by willfrew** [comment]
@richie3366 Sorry for the delay in getting back to you - I just took a look through the nock codebase and, yeah, it seems that we can't easily get access to the raw body.

I think in this instance let's go with your first suggestion to `JSON.stringify` _only_ in the `Content-Length` test then remove the other `parse` calls. It's a bit hacky since it's not really checking the length of what gets sent over the wire but at least it's only limited to that test.

Thanks for your help with this!

**Comment by richie3366** [comment]
No worries about the delay, I'm using my fork until PR is merged.

I hope that it's all good to go.
Let me know if you have any question &/or suggestion.

File: test/rpc.test.js
Comment (willfrew) [inline test/rpc.test.js]: Sorry to be a pain - but could we refer to `requestBody` in the following assertions instead of creating the renaming to `body`? Ditto for the next two changes too

**Review by willfrew — CHANGES_REQUESTED** [review]
Minor change but ready to merge after that!

File: test/rpc.test.js
Comment (richie3366) [reply test/rpc.test.js]: Oh yep, you're right.
I was too much focused on solving the issue that I forgot to read the code again.

**Review by willfrew — APPROVED** [review]
Legend, thanks very much for the change & iterations!

**Comment by willfrew** [comment]
Pushed this out as v1.1.0

**Pull request merged**
